core/primitives-core/src/config.rs

```diff
@@ -256,337 +258,114 @@ pub struct ViewConfig {
-            base: 0,
-            contract_loading_base: 0,
-            contract_loading_bytes: 0,
-            read_memory_base: 0,
-            read_memory_byte: 0,
-            write_memory_base: 0,
-            write_memory_byte: 0,
-            read_register_base: 0,
-            read_register_byte: 0,
-            write_register_base: 0,
-            write_register_byte: 0,
-            utf8_decoding_base: 0,
-            utf8_decoding_byte: 0,
-            utf16_decoding_base: 0,
-            utf16_decoding_byte: 0,
-            sha256_base: 0,
-            sha256_byte: 0,
-            keccak256_base: 0,
-            keccak256_byte: 0,
-            keccak512_base: 0,
-            keccak512_byte: 0,
-            ripemd160_base: 0,
-            ripemd160_block: 0,
-            #[cfg(feature = "protocol_feature_ed25519_verify")]
-            ed25519_verify_base: 0,
-            #[cfg(feature = "protocol_feature_ed25519_verify")]
-            ed25519_verify_byte: 0,
-            ecrecover_base: 0,
-            log_base: 0,
-            log_byte: 0,
-            storage_write_base: 0,
-            storage_write_key_byte: 0,
-            storage_write_value_byte: 0,
-            storage_write_evicted_byte: 0,
-            storage_read_base: 0,
-            storage_read_key_byte: 0,
-            storage_read_value_byte: 0,
-            storage_remove_base: 0,
-            storage_remove_key_byte: 0,
-            storage_remove_ret_value_byte: 0,
-            storage_has_key_base: 0,
-            storage_has_key_byte: 0,
-            storage_iter_create_prefix_base: 0,
-            storage_iter_create_prefix_byte: 0,
-            storage_iter_create_range_base: 0,
-            storage_iter_create_from_byte: 0,
-            storage_iter_create_to_byte: 0,
-            storage_iter_next_base: 0,
-            storage_iter_next_key_byte: 0,
-            storage_iter_next_value_byte: 0,
-            touching_trie_node: 0,
-            read_cached_trie_node: 0,
-            promise_and_base: 0,
-            promise_and_per_promise: 0,
-            promise_return: 0,
-            validator_stake_base: 0,
-            validator_total_stake_base: 0,
-            _unused1: 0,
-            _unused2: 0,
-            alt_bn128_g1_multiexp_base: 0,
-            alt_bn128_g1_multiexp_element: 0,
-            alt_bn128_pairing_check_base: 0,
-            alt_bn128_pairing_check_element: 0,
-            alt_bn128_g1_sum_base: 0,
-            alt_bn128_g1_sum_element: 0,
+            costs: enum_map! {
+                _ => 0
+            },
         }
     }
 }
 
 /// Strongly-typed representation of the fees for counting.
 #[derive(
-    Copy, Clone, Hash, PartialEq, Eq, Debug, PartialOrd, Ord, EnumCount, Display, strum::EnumIter,
+    Copy,
+    Clone,
+    Hash,
+    PartialEq,
+    Eq,
+    Debug,
+    PartialOrd,
+    Ord,
+    EnumCount,
+    Display,
+    strum::EnumIter,
+    enum_map::Enum,
 )]
 #[allow(non_camel_case_types)]
 pub enum ExtCosts {
```

Comment:
Not related to the PR, but I think `ExtCost` is a better name, because a single entity corresponds to a single cost, not many of them.

Reply:
Yes it has been bothering me ever since I joined haha. Same for ActionCosts. I'm just not a huge fan of renaming things, as it messes with git history badly.